### ISSUE TITLE
Checkbox value empty

### ISSUE THREAD

**Issue by jraack** [getgrav/grav-plugin-form#343]
Hi all,

I was updating grav to the latest version, and the form plugin as well.
Now checkboxes are not saved anymore. Previously it was saving the label, with yes/no value.

```

---
title: A Page with an Example Form
form:
    name: contact-form
    fields:
        - name: name
          label: Name
          placeholder: Enter your name
          autofocus: on
          autocomplete: on
          type: text
          validate:
            required: true

        - name: email
          label: Email
          placeholder: Enter your email address
          type: email
          validate:
            required: true
        - name: test
          label: I agree
          type: checkbox
    buttons:
        - type: submit
          value: Submit
        - type: reset
          value: Reset

    process:
        - save:
            fileprefix: feedback-
            dateformat: Ymd-His-u
            extension: txt
            body: "{% include 'forms/data.txt.twig' %}"
        - message: Thank you for your feedback!
        - display: thankyou

---

# My Form

Regular **markdown** content goes here...

```

When i submit, it saves only the field name, with an empty value.

Is my yaml correct?

Thanks in advance.

**Comment by rhukster** [comment]
Think this one requires this Grav fix: 

https://github.com/getgrav/grav/commit/0acb38f5867606409c860304c4711172c7de3a5d

**Comment by jraack** [comment]
@rhukster Thanks! that was exactly the problem.

**Comment by Jimmybow123** [comment]
> Think this one requires this Grav fix:
> 
> [getgrav/grav@0acb38f](https://github.com/getgrav/grav/commit/0acb38f5867606409c860304c4711172c7de3a5d)

Is I am currently having the same problem, where the value of a checkbox on my form isn't being saved, and only a null value is being returned. I checked and I have grav version 1.6.9, which has the code change specified above. Any thoughts?

**Comment by mahagr** [comment]
Do you have a small example on how to reproduce this issue of yours?

**Comment by drzraf** [comment]
The behavior is:

With **default: true**
```yaml
        ap:
            label: foo
            type: checkbox
            id: ap_field
            default: true
```


```html
        <label class="form-checkbox" for="ap_field-">
        <input name="ap" value="1" type="checkbox" id="ap_field-" checked="checked" class="">
            <div class="box"></div>
            <div class="label">foo</div>
        </label>
```

Sounds ok.

With **default: false**
```yaml
        ap:
            label: foo
            type: checkbox
            id: ap_field
            default: false
```

```html
        <label class="form-checkbox" for="ap_field-">
        <input name="ap" value="" type="checkbox" id="ap_field-" class="">
            <div class="box"></div>
            <div class="label">foo</div>
        </label>
```

In that case, if that checkbox is checked and form submitted, then the backend (Grav or something else) receives `ap=` (empty value). It would be preferable to receive `ap=1` or `ap=on`.

Basically:
- checkbox value is inconsistent (because it depends upon the checked default status)
- We miss control over its `value`.

**Comment by ricardo118** [comment]
gravs form fields have name="data[ap]" did u do the custom form html and not use data?

**Comment by drzraf** [comment]
I've a plugin that set `scope` to `''` because I don't want this `data.` namespace when submitting to remote endpoint.

**Comment by ricardo118** [comment]
my guess is because of that, something isnt saving as grav is expecting that scope

**Comment by drzraf** [comment]
But the problem is not that the form's checkbox isn't saved by Grav because I'm sending the form to Grav but somewhere else.
The problem is that I can't control the value of the checkbox (the `value` is empty).

**Comment by w00fz** [comment]
When a checkbox is unchecked and you submit a form, the standard behavior is for the value to be empty.
Value is only sent when the checkbox is checked.

From MDN:

> Note: If a checkbox is unchecked when its form is submitted, there is no value submitted to the server to represent its unchecked state (e.g. value=unchecked); the value is not submitted to the server at all. If you wanted to submit a default value for the checkbox when it is unchecked, you could include an <input type="hidden"> inside the form with the same name and value, generated by JavaScript perhaps.


https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#Value


There might be ways to work around this but you might have to customize the behavior yourself. 
Maybe you can also have a look at the checkbox and override it to behave like you are expecting. 

I don't think this is an issue though.

**Comment by mahagr** [comment]
Another tip: use strings/integers and not booleans as the default values. Boolean gets converted to 1 (true) and '' (false) by PHP unless you use the new (and fixed) `|string` filter in Grav 1.7.

**Comment by drzraf** [comment]
I just test the latest 4.0.10 and it respects `value` and `default` parameters. I think it could be closed.
